src/EnergyPlus/EarthTube.cc

```diff
@@ -543,6 +631,188 @@ void CheckEarthTubesInZones(EnergyPlusData &state,
+                SetupOutputVariable(state,
+                                    format("Earth Tube Node Temperature {}", nodeNum),
+                                    OutputProcessor::Unit::C,
+                                    thisEarthTube.tCurrent[nodeNum - 1],
+                                    OutputProcessor::SOVTimeStepType::Zone,
+                                    OutputProcessor::SOVStoreType::State,
+                                    zone.Name);
+                SetupOutputVariable(state,
+                                    format("Earth Tube Undisturbed Ground Temperature {}", nodeNum),
+                                    OutputProcessor::Unit::C,
+                                    thisEarthTube.tUndist[nodeNum - 1],
+                                    OutputProcessor::SOVTimeStepType::Zone,
+                                    OutputProcessor::SOVStoreType::State,
+                                    zone.Name);
+            }
+            SetupOutputVariable(state,
+                                "Earth Tube Upper Boundary Ground Temperature",
+                                OutputProcessor::Unit::C,
+                                thisEarthTube.tUpperBound,
+                                OutputProcessor::SOVTimeStepType::Zone,
+                                OutputProcessor::SOVStoreType::State,
+                                zone.Name);
+            SetupOutputVariable(state,
+                                "Earth Tube Lower Boundary Ground Temperature",
+                                OutputProcessor::Unit::C,
+                                thisEarthTube.tLowerBound,
+                                OutputProcessor::SOVTimeStepType::Zone,
+                                OutputProcessor::SOVStoreType::State,
+                                zone.Name);
+        }
+    } // ...end of firstTimeInits block
+
+    if (state.dataGlobal->BeginDayFlag || state.dataGlobal->BeginEnvrnFlag) {
+        // update all of the undisturbed temperatures (only need to do this once per day because the equation only changes as the day changes
+        for (int etNum = 1; etNum <= totEarthTube; ++etNum) {
+            auto &thisEarthTube = state.dataEarthTube->EarthTubeSys(etNum);
+            if (thisEarthTube.ModelType != EarthTubeModelType::Vertical) continue; // Skip earth tubes that do not use vertical solution
+            thisEarthTube.tUpperBound = thisEarthTube.calcUndisturbedGroundTemperature(state, thisEarthTube.depthUpperBound);
+            thisEarthTube.tLowerBound = thisEarthTube.calcUndisturbedGroundTemperature(state, thisEarthTube.depthLowerBound);
+            for (int nodeNum = 0; nodeNum <= thisEarthTube.totNodes - 1; ++nodeNum) {
+                thisEarthTube.tUndist[nodeNum] = thisEarthTube.calcUndisturbedGroundTemperature(state, thisEarthTube.depthNode[nodeNum]);
+            }
+        }
+    } // ...end of BeginDayFlag block
```

Comment:
Will this get hit multiple times while BeginDayFlag is true?  Do you need a second flag like "MyBeginDayFlag" so that it only happens once per day?

Reply:
I'll double check.  I was trying to avoid a duplicate calculation but I may have missed something.

Reply:
Yeah, this was a problem.  I took a page out of the ElectricPowerServiceManager.cc and decided to use "timeElapsed" to avoid duplicate initializations and improper updates.  Will be testing this out here shortly.